### ISSUE TITLE
0-bootstrap failing at repo setup in gar and cloud build triggers

### ISSUE THREAD

**Issue by Atul7696** [terraform-google-modules/terraform-example-foundation#723]
### TL;DR

0-bootstrap failing at repo setup in gar and cloud build triggers

### Expected behavior

All the resources should have been deployed. However, it fails at cloud build part where it fails with below message. Also terraform validator also fails.

It does create all other things like folder, projects for seed and cicd. Enable APIS, create service accounts, add them to related roles. 

### Observed behavior

All the resources should have been deployed. However, it fails at cloud build part where it fails with below message. Also terraform validator also fails.

### Terraform Configuration

```hcl
Link to terraform JSON plan file
https://drive.google.com/file/d/1ilN9hIVUrFQcZ_fOeiKh4od7rfHf6wJc/view?usp=sharing

Using default cloud foundation toolkit  and using 0-bootstrap folder with making changes to terraform.tfvars filde and deploying via terraform plan and following the documentation

### References

https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/0-bootstrap
```

### Terraform Version

```sh
1.2.1
```

### Additional information

Terraform validator gives below error

022-06-01T00:27:28.553200017Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO] Authenticating using DefaultClient...", "context": ""}}
2022-06-01T00:27:28.553310874Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]", "context": ""}}
2022-06-01T00:27:28.556015766Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO] Authenticating using DefaultClient...", "context": ""}}
2022-06-01T00:27:28.556096902Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]", "context": ""}}
2022-06-01T00:27:28.557632362Z  info    {"version": "v1.0.0", "error_details": {"error": "[DEBUG] Waiting for state to become: [success]", "context": ""}}
2022-06-01T00:27:28.625216021Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO] Terraform is using this identity: atulXXXX@gmail.com", "context": ""}}
2022-06-01T00:27:28.625443965Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO] Instantiating Google Cloud ResourceManager client for path https://cloudresourcemanager.googleapis.com/", "context": ""}}
2022-06-01T00:27:28.639586163Z  error   {"version": "v1.0.0", "error_details": {"error": "converting tfplan to CAI assets: reading resource changes: reading JSON plan: invalid character 'ÿ' looking for beginning of value", "context": "github.com/GoogleCloudPlatform/terraform-validator/cmd.Execute\n\t/Users/stephenrlewis/projects/terraform-validator/cmd/root.go:111\nmain.main\n\t/Users/stephenrlewis/projects/terraform-validator/main.go:30\nruntime.main\n\t/usr/local/Cellar/go/1.16.6/libexec/src/runtime/proc.go:225"}}

When i ran terraform validator post applying terraform it gave below 

terraform-validator validate test.json --policy-path="../policy-library" --project project-k8-271708
2022-06-01T00:56:08.782108377Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO] Authenticating using DefaultClient...", "context": ""}}
2022-06-01T00:56:08.782229132Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]", "context": ""}}
2022-06-01T00:56:08.784688433Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO] Authenticating using DefaultClient...", "context": ""}}
2022-06-01T00:56:08.78478427Z   info    {"version": "v1.0.0", "error_details": {"error": "[INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]", "context": ""}}
2022-06-01T00:56:08.786664651Z  info    {"version": "v1.0.0", "error_details": {"error": "[DEBUG] Waiting for state to become: [success]", "context": ""}}
2022-06-01T00:56:08.853861431Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO] Terraform is using this identity: atul7696@gmail.com", "context": ""}}
2022-06-01T00:56:08.854168361Z  info    {"version": "v1.0.0", "error_details": {"error": "[INFO] Instantiating Google Cloud ResourceManager client for path https://cloudresourcemanager.googleapis.com/", "context": ""}}
2022-06-01T00:56:08.883376858Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_billing_account_iam_member", "context": ""}}
2022-06-01T00:56:08.88343319Z   info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_folder", "context": ""}}
2022-06-01T00:56:08.883446961Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_sourcerepo_repository", "context": ""}}
2022-06-01T00:56:08.883455435Z  info    {"version": "v1.0.0", "error_details": {"error": "unknown resource: google_artifact_registry_repository", "context": ""}}
2022-06-01T00:56:08.883462987Z  info    {"version": "v1.0.0", "error_details": {"error": "unknown resource: google_artifact_registry_repository_iam_member", "context": ""}}
2022-06-01T00:56:08.883486281Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_cloudbuild_trigger", "context": ""}}
2022-06-01T00:56:08.883493972Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_cloudbuild_trigger", "context": ""}}
2022-06-01T00:56:08.883501421Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_cloudbuild_trigger", "context": ""}}
2022-06-01T00:56:08.883513745Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_cloudbuild_trigger", "context": ""}}
2022-06-01T00:56:08.883524356Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_cloudbuild_trigger", "context": ""}}
2022-06-01T00:56:08.883532701Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_cloudbuild_trigger", "context": ""}}
2022-06-01T00:56:08.883539644Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_cloudbuild_trigger", "context": ""}}
2022-06-01T00:56:08.883550095Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_cloudbuild_trigger", "context": ""}}
2022-06-01T00:56:08.883561386Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_service_account_iam_member", "context": ""}}
2022-06-01T00:56:08.883572154Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_sourcerepo_repository", "context": ""}}
2022-06-01T00:56:08.883584967Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_sourcerepo_repository", "context": ""}}
2022-06-01T00:56:08.883596274Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_sourcerepo_repository", "context": ""}}
2022-06-01T00:56:08.883607677Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_sourcerepo_repository", "context": ""}}
2022-06-01T00:56:08.88362057Z   info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_project_default_service_accounts", "context": ""}}
2022-06-01T00:56:08.883633364Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_service_account", "context": ""}}
2022-06-01T00:56:08.883649766Z  info    {"version": "v1.0.0", "error_details": {"error": "unknown resource: random_id", "context": ""}}
2022-06-01T00:56:08.883665685Z  info    {"version": "v1.0.0", "error_details": {"error": "unknown resource: null_resource", "context": ""}}
2022-06-01T00:56:08.883673253Z  info    {"version": "v1.0.0", "error_details": {"error": "unknown resource: random_id", "context": ""}}
2022-06-01T00:56:08.88368085Z   info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_billing_account_iam_member", "context": ""}}
2022-06-01T00:56:08.883693967Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_service_account", "context": ""}}
2022-06-01T00:56:08.883705891Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_service_account_iam_member", "context": ""}}
2022-06-01T00:56:08.883735068Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_project_default_service_accounts", "context": ""}}
2022-06-01T00:56:08.883763877Z  info    {"version": "v1.0.0", "error_details": {"error": "unsupported resource: google_resource_manager_lien", "context": ""}}
2022-06-01T00:56:08.883783384Z  info    {"version": "v1.0.0", "error_details": {"error": "unknown resource: random_id", "context": ""}}
2022-06-01T00:56:08.883794805Z  info    {"version": "v1.0.0", "error_details": {"error": "unknown resource: random_id", "context": ""}}
No violations found.

### References

https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/0-bootstrap 

**Error while running terraform apply again**

╷
│ Error: local-exec provisioner error
│
│ with module.cloudbuild_bootstrap.null_resource.cloudbuild_terraform_builder,
│ on .terraform/modules/cloudbuild_bootstrap/modules/cloudbuild/main.tf line 194, in resource "null_resource" "cloudbuild_terraform_builder":
│ 194: provisioner "local-exec" {
│
│ Error running command ' gcloud builds submit
│ .terraform/modules/cloudbuild_bootstrap/modules/cloudbuild/cloudbuild_builder/ --project
│ prj-b-cicd-22dd
│ --config=.terraform/modules/cloudbuild_bootstrap/modules/cloudbuild/cloudbuild_builder/cloudbuild.yaml
│ --substitutions=_TERRAFORM_VERSION=1.2.1,_TERRAFORM_VERSION_SHA256SUM=4a52886e019b4fdad2439da5ff43388bbcc6cce9784fde32c53dcd0e28ca9957,_TERRAFORM_VALIDATOR_RELEASE=v0.6.0,_REGION=us-east4,_REPOSITORY=prj-tf-runners
│ ': exit status 1. Output: Creating temporary tarball archive of 4 file(s) totalling 6.1 KiB
│ before compression.
│ Uploading tarball of
│ [.terraform/modules/cloudbuild_bootstrap/modules/cloudbuild/cloudbuild_builder/] to
│ [gs://prj-b-cicd-22dd_cloudbuild/source/1654068826.048646-7f37b2da8be04509b7393f4a5f40d5b8.tgz]
│ ERROR: (gcloud.builds.submit) INVALID_ARGUMENT: invalid build: invalid build step name
│ "us-east4-docker.pkg.dev/prj-b-cicd-22dd/prj-tf-runners\r/terraform": could not parse reference:
/terraform-docker.pkg.dev/prj-b-cicd-22dd/prj-tf-runners

**Comment by Atul7696** [comment]
Used a Linux VM as GCE instance and deployed from there. It got successful. Artifact registry also got created. Please advise does it also create cloud build triggers. Don't see any though.

Also when I am trying to push the 1-org/environments/shared repo tp plan and production branch, Code gets pushed but no output is shown in cloud build. Can somebody advise here? @daniel-cit Please suggest

**Comment by mlutx** [comment]
I was able to run `terraform apply` on a Ubuntu system with no error reported in 0-bootstrap step, and I can see the builds and triggers in the Terraform statefile, for example:

`~/gcp/terraform-example-foundation/0-bootstrap$ terraform state show module.cloudbuild_bootstrap.google_cloudbuild_trigger.main_trigger[\"gcp-org\"]
# module.cloudbuild_bootstrap.google_cloudbuild_trigger.main_trigger["gcp-org"]:
resource "google_cloudbuild_trigger" "main_trigger" {
    create_time    = "2022-07-06T22:21:00.928077299Z"
    description    = "gcp-org - terraform apply."
    disabled       = false
    filename       = "cloudbuild-tf-apply.yaml"
    id             = "projects/prj-b-cicd-d0d2/triggers/3f2b388c-263b-4c4e-ac5b-52f99a0fc56e"
    ignored_files  = []
    included_files = []
    name           = "gcp-org---terraform-apply"
    project        = "prj-b-cicd-d0d2"
    substitutions  = {
        "_ARTIFACT_BUCKET_NAME" = "prj-cloudbuild-artifacts-7a82"
        "_BILLING_ID"           = "xxxxxx-xxxxxx-xxxxxx"
        "_DEFAULT_REGION"       = "us-central1"
        "_GAR_REPOSITORY"       = "prj-tf-runners"
        "_ORG_ID"               = "123456789012"
        "_STATE_BUCKET_NAME"    = "bkt-b-tfstate-0f56"
        "_TF_ACTION"            = "apply"
        "_TF_SA_EMAIL"          = "org-terraform@prj-b-seed-xxxx.iam.gserviceaccount.com"
    }
    tags           = []
    trigger_id     = "3f2b388c-263b-4c4e-ac5b-52f99a0fc56e"

    approval_config {
        approval_required = false
    }

    trigger_template {
        branch_name  = "^(development|non\\-production|production)$"
        invert_regex = false
        project_id   = "prj-b-cicd-d0d2"
        repo_name    = "gcp-org"
    }
}
`

I don't see any builds and triggers from the console or from gcloud output. After pushing from 1-org there's nothing triggered in cloud build either. 

Is there anything missing here?

**Comment by Atul7696** [comment]
Closing as issue got resolved
